ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngine.kt

```diff
@@ -31,6 +32,12 @@ interface HttpClientEngine : CoroutineScope, Closeable {
      */
     val config: HttpClientEngineConfig
 
+    /**
+     * Set of supported engine extensions.
+     */
+    @KtorExperimentalAPI
+    val supportedCapabilities: Set<HttpClientEngineCapability<*>>
```

Comment:
Shouldn't we have default empty set implementation here?

Reply:
It's an interface, so we can't set a default value here. Do you know how to bypass it?

Reply:
Thanks, done.